File: LeetCode/575. Distribute Candies.cpp
class Solution {
public:
    int distributeCandies(vector<int>& candies) {
        int sz = (int)candies.size();
        sort(candies.begin(),candies.end());
        int dis = sz>0;
        for(int i = 1 ;i<sz;i++){
            dis+=candies[i]!=candies[i-1];
        }
        return min(sz>>1,dis);
    }
};

File: LeetCode/575. Distribute Candies.py
class Solution(object):
    def distributeCandies(self, candies):
        candies.sort()
        dis = 0
        sz = len(candies)
        i = 1
        if sz>1:
            dis+=1
        while i<sz:
            if candies[i]!=candies[i-1]:
                dis+=1
            i+=1
        return min(dis,sz>>1)     
        
